web/views/bundles_view.ex

```diff
@@ -1,36 +1,45 @@
 defmodule Cog.V1.BundlesView do
   use Cog.Web, :view
 
-  def render("bundle.json", %{bundle: bundle}) do
+  alias Cog.V1.RelayGroupView
+  alias Cog.V1.RuleView
+
+  def render("bundle.json", %{bundle: bundle}=params) do
     %{id: bundle.id,
       name: bundle.name,
-      commands: render_commands(bundle.commands),
-      relay_groups: render_relay_groups(bundle.relay_groups),
       enabled: bundle.enabled,
       inserted_at: bundle.inserted_at,
       updated_at: bundle.updated_at}
+    |> Map.merge(render_includes(params, bundle))
   end
+
   def render("index.json", %{bundles: bundles}) do
-    %{bundles: render_many(bundles, __MODULE__, "bundle.json", as: :bundle)}
+    %{bundles: render_many(bundles, __MODULE__, "bundle.json", as: :bundle, include: [:commands, :relay_groups])}
   end
+
   def render("show.json", %{bundle: bundle}) do
-    %{bundle: render_one(bundle, __MODULE__, "bundle.json", as: :bundle)}
+    %{bundle: render_one(bundle, __MODULE__, "bundle.json", as: :bundle, include: [:commands, :relay_groups])}
   end
 
-  def render_relay_groups(groups) do
-    for group <- groups do
-      %{id: group.id,
-        name: group.name}
-    end
+  def render("command.json", %{command: command}) do
+    %{id: command.id,
+      name: command.name,
+      documentation: command.documentation,
+      enforcing: command.enforcing,
+      execution: command.execution,
+      rules: render_many(command.rules, RuleView, "rule.json")}
   end
 
-  def render_commands(commands) do
-    for command <- commands do
-      %{id: command.id,
-        name: command.name,
-        documentation: command.documentation,
-        enforcing: command.enforcing,
-        execution: command.execution}
-    end
+  defp render_includes(params, bundle) do
+    Map.get(params, :include, [])
+    |> Enum.reduce(%{}, fn(inc, reply) -> Map.put(reply, inc, render_include(inc, bundle)) end)
   end
+
+  defp render_include(:commands, bundle) do
+    render_many(bundle.commands, __MODULE__, "command.json", as: :command)
+  end
+  defp render_include(:relay_groups, bundle) do
+    render_many(bundle.relay_groups, RelayGroupView, "relay_group.json")
+  end
```

Comment:
Nope, that happens a couple of lines up in the `render_includes` function that calls the individual includes and merges them into the reply map.